### PR TITLE
BaSP-MR-62: Trainers components added

### PR DESCRIPTION
**Pull request by MatiasSauvanet**
Features:

-Divide index in two components (Index and formTrainer)
-Still need some visuals
-Added classmates components
-Added a TimeOut to the add and edit success modal

### CODE REVIEW

File: src/Components/Trainers/FormTrainers/index.jsx [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  const getTrainerID = async (id) => {
+    try {
+      const res = await fetch(`${process.env.REACT_APP_API_URL}/api/trainers/${id}`);
+      const { data, error, message } = await res.json();
+      if (data) {
+        setInputs(data);
+      }
+      console.log(error);
+      console.log(message);
+      return data;
```
Comment (Bigzufi) [inline src/Components/Trainers/FormTrainers/index.jsx:33]: Remove console.log

File: src/Components/Trainers/FormTrainers/index.jsx [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        method: 'PUT',
+        body: JSON.stringify(setEditTrainer),
+        headers: {
+          'Content-Type': 'application/json'
+        }
+      });
+      const { message, error, data } = await response.json();
+      console.log(data);
+      if (!error) {
+        console.log(message);
```
Comment (Bigzufi) [inline src/Components/Trainers/FormTrainers/index.jsx:60]: Remove console.log

File: src/Components/Trainers/FormTrainers/index.jsx [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+      setTimeout(() => {
+        onRedirect.state.message = data.message;
+        history.push(onRedirect);
+      }, 1500);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+  const sendTrainer = async (item) => {
+    console.log(item);
```
Comment (Bigzufi) [inline src/Components/Trainers/FormTrainers/index.jsx:73]: Remove console.log

File: src/Components/Trainers/FormTrainers/index.jsx [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+        method: 'POST',
+        body: JSON.stringify(item),
+        headers: {
+          'Content-Type': 'application/json'
+        }
+      });
+      const { data, message, error } = await response.json();
+      if (!error) {
+        console.log(data);
+      } else {
```
Comment (Bigzufi) [inline src/Components/Trainers/FormTrainers/index.jsx:85]: Remove console.log

**Comment by Bigzufi** [comment]
The delete function is not working correctly. It doesn't delete the selected trainer but removes the last one from the list instead. The add trainer function is also not working properly. In case of a validation error, it still displays a success message, and the trainer is not created. Additionally, you should resolve conflicts.

File: src/Components/Trainers/index.jsx [hunk 5]
```diff
@@ -28,285 +19,101 @@ const Trainers = () => {
...
-        setToggleEdit(false);
+        setTrainers([...trainers.filter((trainer) => trainer._id !== _id)]);
       } else {
         throw message;
       }
+      setDeleteModal(!deleteModal);
+      setSuccessModal(!successModal);
     } catch (error) {
-      console.error(error);
+      alert(error);
```
Comment (DelfiLT) [inline src/Components/Trainers/index.jsx:35]: Don't use alerts

File: src/Components/Trainers/FormTrainers/index.jsx [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+        headers: {
+          'Content-Type': 'application/json'
+        }
+      });
+      const { message, error, data } = await response.json();
+      console.log(data);
+      if (!error) {
+        console.log(message);
+      } else {
+        throw message;
```
Comment (DelfiLT) [inline src/Components/Trainers/FormTrainers/index.jsx:62]: this isn't working, you should report this message on the modal

File: src/Components/Trainers/FormTrainers/index.jsx [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+        body: JSON.stringify(item),
+        headers: {
+          'Content-Type': 'application/json'
+        }
+      });
+      const { data, message, error } = await response.json();
+      if (!error) {
+        console.log(data);
+      } else {
+        throw message;
```
Comment (DelfiLT) [inline src/Components/Trainers/FormTrainers/index.jsx:86]: Same here

**Review by DelfiLT — CHANGES_REQUESTED** [review]
Correct the changes requested and for the UI you could make the table a little bigger and add a padding to the blue box to look prettier, also, you should fix the font size from the table.

**Review by DelfiLT — APPROVED** [review]

**Review by Dannylez — APPROVED** [review]

**Pull request merged**